datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -2125,6 +2141,15 @@ mod test {
             ),
         )))?;
 
+        let struct_field_1 = Field::new("c0", DataType::Boolean, true);
+        let struct_field_2 = Field::new("c1", DataType::Int32, true);
+        round_trip_literal(
```

Comment:
I suggest one more test with null values too -- specifically something like

```rust
            ScalarStructBuilder::new()
                .with_scalar(struct_field_1, ScalarValue::Boolean(None))
                .with_scalar(struct_field_2, ScalarValue::Int32(None))
                .build()?,
```

Reply:
Added (though incorporated into the existing test) in [84ccaa3](https://github.com/apache/datafusion/pull/10622/commits/84ccaa3361f64dc4a247487b43a57367734b9548)!